tests/unit/test_configuration.py

```diff
@@ -1,21 +1,44 @@
+from pathlib import Path
+
 import pytest
 from vivarium.config_tree import ConfigTree
 
 from pseudopeople.entities import Form
-from pseudopeople.utilities import get_default_configuration
+from pseudopeople.utilities import (
+    get_default_configuration,
+    update_configuration_with_yaml,
+)
+
+
+@pytest.fixture
+def user_configuration_yaml(tmp_path):
+    text = """decennial_census:
+    omission: 0.05
+    first_name:
+        nickname:
+            row_noise_level: 0.05"""
+    user_config_path = Path(f"{tmp_path}/test_configuration.yaml")
+    with open(user_config_path, "w") as file:
+        file.write(text)
+    return user_config_path
 
 
 def test_default_configuration():
     config = get_default_configuration()
     assert config
     assert isinstance(config, ConfigTree)
     # TODO: From Rajan: We should test that this configuration actually matches
-    # what we'd expect it to be. We can do this either by comparing it to the
-    # values in the yaml file, or by just confirming that the correct call to
-    # config_tree.update() was made in the function. The latter seems preferable
-    # to me as a unit test.
+    #  what we'd expect it to be. We can do this either by comparing it to the
+    #  values in the yaml file, or by just confirming that the correct call to
+    #  config_tree.update() was made in the function. The latter seems preferable
+    #  to me as a unit test.
 
 
-@pytest.mark.skip(reason="TODO")
-def test_user_configuration_file():
-    pass
+def test_user_configuration_file(user_configuration_yaml):
+    """Test that a user config yaml will override the default configuration."""
+    config = get_default_configuration()
+    config = update_configuration_with_yaml(config, user_configuration_yaml)
+    assert (
```

Comment:
This test is failing and I'm not really sure what it is intending to test. Why are we comparing the "omission" config value to the "nickname" config value? 

As I mentioned in the note that is in the TODO above, we can either test our configuration is correct by comparing the values of keys in the configuration to the values we expect or by trusting that `ConfigTree` works as specified and merely check that we have made the correct calls to create and update the `ConfigTree`.

Reply:
We spoke in person about this but to circle back on this. It was to test just a subset of values I was expecting to replace. I would like to go the latter route you suggest and am trying to work out how to mock things to count the `update()` calls.